File: Week 2/Task 4/Cross-Language Translation/ProcessData.kt
fun processData(records: List<Map<String, Any>>): Map<Any, Double> {
    val result = mutableMapOf<Any, Double>()
    for (record in records) {
        try {
            if (record["status"] != "active") continue

            val value = (record["value"] as? Number)?.toDouble()
                ?: (record["value"] as? String)?.toDoubleOrNull()
                ?: throw IllegalArgumentException("Value is not a number")

            if (value < 0) throw IllegalArgumentException("Negative value not allowed")

            val id = record["id"] ?: throw IllegalArgumentException("Missing id")
            result[id] = value * 2
        } catch (e: Exception) {
            println("Skipping record due to error: ${e.message}")
        }
    }
    return result
} 

File: Week 2/Task 4/Cross-Language Translation/process_data.py
def process_data(records):
    """
    Processes a list of records, filtering and transforming data.
    Args:
        records (list of dict): Each dict contains 'id', 'value', and 'status'.
    Returns:
        dict: Mapping of 'id' to processed 'value' for valid records.
    """
    result = {}
    for record in records:
        try:
            if record['status'] != 'active':
                continue
            value = float(record['value'])
            if value < 0:
                raise ValueError("Negative value not allowed")
            result[record['id']] = value * 2
        except (KeyError, ValueError) as e:
            print(f"Skipping record due to error: {e}")
    return result 